ambry-api/src/main/java/com.github.ambry/config/StoreConfig.java

```diff
@@ -118,6 +118,20 @@
   @Default("50")
   public final int storeMinUsedCapacityToTriggerCompactionInPercentage;
 
+  /**
+   * The factory class used to get the compaction policy
+   */
+  @Config("store.compaction.policy.factory")
+  @Default("com.github.ambry.store.DefaultCompactionPolicyFactory")
+  public final String storeCompactionPolicyFactory;
+
+  /**
+   * The minimum number of log segments to be reclaimed to trigger compaction
+   */
+  @Config("store.min.log.segment.count.to.reclaim.to.trigger.compaction")
+  @Default("1")
+  public final int storeMinLogSegmentCountToReclaimToTriggerCompaction;
```

Comment:
How do we make it clear that this need not be followed by all compaction policies? Perhaps you can mention that in the documentation that respecting this config is up to the implementation of the individual policy impls